storage/innobase/include/trx0purge.h

```diff
@@ -432,10 +434,17 @@ class purge_sys_t
 
   struct view_guard
   {
-    inline view_guard();
+    /** latch type: 0=view with no latch; -1=end_view, 1=view */
+    const int latch;
+    inline view_guard(int latch);
```

Comment:
Outside the purge tasks, we must protect the undo log pages either by `purge_sys.latch` or `purge_sys.end_latch`, depending on which type of MVCC read we are executing. Using separate `view_guard` and `end_view_guard` would have resulted in code duplication, because we are now extending the lifetime of the guard.